### PR TITLE
Stop using Object.assign.

### PR DESCRIPTION
**Pull request by tamuratak**
Stop using Object.assign.

### CODE REVIEW

**Comment by jlelong** [comment]
We should also replace https://github.com/James-Yu/LaTeX-Workshop/blob/dfaba4137c555aa785ab6f2644f25dae80cafdbe/src/providers/completer/glossary.ts#L46-L47 by

```js
let keys = Array.from(new Set(suggestions.keys())
```

as `Object.keys` does not return the keys of the` Map`.

File: src/providers/completer/glossary.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ export class Glossary implements IProvider {
         if (result[1] && result[1].match(/^ac/i)) {
             suggestions = this.acronyms
         } else {
-            suggestions = Object.assign({}, this.acronyms, this.glossaries)
+            suggestions = new Map( [...this.acronyms.entries(), ...this.glossaries.entries()] )
```
Comment (jlelong) [inline src/providers/completer/glossary.ts:42]: According to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#cloning_and_merging_maps, we can directly write

```js
suggestions = new Map( [...this.acronyms, ...this.glossaries] )
```

**Comment by tamuratak** [comment]
It seems that we can change the whole part

https://github.com/James-Yu/LaTeX-Workshop/blob/365e728b68f050e967c78c4425e1dd974fc49759/src/providers/completer/glossary.ts#L46-L57

to 

```ts
  const items = Array.from(suggestions.values())
  return items
```

Is this correct?

**Comment by jlelong** [comment]
Yes, I agree

**Comment by tamuratak** [comment]
Can I merge this PR?

**Review by jlelong — APPROVED** [review]

**Comment by jlelong** [comment]
Yes

**Pull request merged**